### DISCUSSION TITLE
Can't get collections migration cmd to work

### DISCUSSION THREAD

**Discussion by marcusfelix** [pocketbase/pocketbase#726] [Q&A] [answered]
What i'm doing:

1) Setup collections schema with UI
2) Exported migration file with ```go run main migrate collections```
3) Dockerfile to build PB as framework
4) Build

When i run the image PB opens as a new installation with none of the collections i have created. What i'm doing wrong?

Dockerfile
```dockerfile
FROM golang:1.18-alpine as builder
RUN apk add git
WORKDIR /app
COPY go.mod go.sum ./
RUN go mod download
COPY . .
RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o main .
# --
FROM alpine:latest
RUN apk --no-cache add ca-certificates curl
WORKDIR /app
COPY --from=builder /app/main ./
VOLUME /app/pb_data
EXPOSE 8090
CMD ["./main", "serve", "--http=0.0.0.0:8090"]
```

Thanks in advance

**Comment by ganigeorgiev** [comment, marked as answer]
Have you imported the `migrations` package in your `main.go`? Example:

```go
import _ "yourpackage/migrations"
```

**Reply by marcusfelix** [reply]
oh that was missing :/ thank you

**Reply by marcusfelix** [reply]
@ganigeorgiev when migration run throw ```panic: Failed to apply migration 1664988836_collections_snapshot.go: System collection "profiles" cannot be deleted``` i think it will be better wait users/profile refactoring to generate a new migration file

**Reply by ganigeorgiev** [reply]
This is because probably your project was created before v0.4.x since the `profiles` collection didn't have an explicit id before that. See the notice in https://pocketbase.io/docs/db-migrations/#collections-migration:

> PocketBase allows making Collection changes from the Admin UI and therefore there are some caveats/limitations related to the automated Collections migration:
> 
> - It works only for new projects (v0.4.0+).
> - It is recommended to enable the "Hide collection create and edit controls" option in the Admin UI on the prod environment.
This is because manual Collection changes made from the Admin UI in the migration receiving environment (eg. prod) could conflict with the migration file generated from the local environment since the migration relies on the Collection and Field IDs, which will differ on the two environments if created manually.

You can fix this by changing the id of the `profiles` collection on your local env to match with the id on the prod env or wait for the upcoming v0.8 release.

**Reply by ganigeorgiev** [reply]
@marcusfelix _btw. You can review how the migration will be applied by copying the json from the migration file and paste it inside the Admin UI "Settings > Import collections" on your prod/receiving environment._
